contracts/modules/CollectionOffers/V1/CollectionOfferBookV1.sol

```diff
@@ -0,0 +1,430 @@
+                // Else update is decrease --
+            } else {
+                // Traverse backward until insert location is found
+                _insertDecreasedOffer(_collection, _offerId, _offer.prevId, _newAmount);
+            }
+        }
+    }
+
+    /// @notice Removes an offer from its collection's offer book
+    /// @param _collection The address of the ERC-721 collection
+    /// @param _offerId The ID of the offer
+    function _removeOffer(address _collection, uint32 _offerId) internal {
+        // If offer is only one for collection, remove all associated data
+        if (_isOnlyOffer(_collection, _offerId)) {
+            delete floorOfferId[_collection];
+            delete floorOfferAmount[_collection];
+            delete ceilingOfferId[_collection];
+            delete ceilingOfferAmount[_collection];
+            delete offers[_collection][_offerId];
+
+            // Else if the offer is the current floor, update the collection's floor before removing
+        } else if (_isFloorOffer(_collection, _offerId)) {
+            uint32 newFloorId = offers[_collection][_offerId].nextId;
+            uint256 newFloorAmount = offers[_collection][newFloorId].amount;
+
+            offers[_collection][newFloorId].prevId = 0;
+
+            floorOfferId[_collection] = newFloorId;
+            floorOfferAmount[_collection] = newFloorAmount;
+
+            delete offers[_collection][_offerId];
+
+            // Else if the offer is the current ceiling, update the collection's ceiling before removing
+        } else if (_isCeilingOffer(_collection, _offerId)) {
+            uint32 newCeilingId = offers[_collection][_offerId].prevId;
+            uint256 newCeilingAmount = offers[_collection][newCeilingId].amount;
+
+            offers[_collection][newCeilingId].nextId = 0;
+
+            ceilingOfferId[_collection] = newCeilingId;
+            ceilingOfferAmount[_collection] = newCeilingAmount;
+
+            delete offers[_collection][_offerId];
+
+            // Else offer is in middle, so update its previous and next neighboring pointers before removing
+        } else {
+            Offer memory offer = offers[_collection][_offerId];
+
+            offers[_collection][offer.nextId].prevId = offer.prevId;
+            offers[_collection][offer.prevId].nextId = offer.nextId;
+
+            delete offers[_collection][_offerId];
+        }
+    }
+
+    /// @notice Finds a collection offer to fill
+    /// @param _collection The address of the ERC-721 collection
+    /// @param _minAmount The minimum offer amount valid to match
+    function _getMatchingOffer(address _collection, uint256 _minAmount) internal view returns (uint32) {
+        // If current ceiling offer is greater than or equal to maker's minimum, return its id to fill
+        if (ceilingOfferAmount[_collection] >= _minAmount) {
+            return ceilingOfferId[_collection];
+            // Else return no offer found
+        } else {
```

Comment:
Inspired by [ESLint's `no-else-return` rule](https://eslint.org/docs/rules/no-else-return) and [Palantir's TSLint `unnecessary-else` rule](https://palantir.github.io/tslint/rules/unnecessary-else/) -- since the `if` block contains a `return` statement, the `else` block becomes unnecessary.

Suggested change:


```solidity
    function _getMatchingOffer(address _collection, uint256 _minAmount) internal view returns (uint32) {
        // If current ceiling offer is greater than or equal to maker's minimum, return its id to fill
        if (ceilingOfferAmount[_collection] >= _minAmount) {
            return ceilingOfferId[_collection];
        }
        
        // Else return no offer found
        return 0;
    }
```


Related: https://github.com/ourzora/v3/pull/141#discussion_r810610919